### PR TITLE
[tedious] Update class definitions

### PR DESCRIPTION
**Pull request by lohart13**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/tediousjs/tedious/blob/master/src/tedious.ts
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### Notes:
This PR is a breaking change to the `@types/tedious` package in which I have used the typescript files in the tedious repo to generate the updated typings.
There are quite a few breaking changes with the classes so I've updated the version to match the release that I used to generate the types.

### CODE REVIEW

**Comment by dhensby** [comment]
I'm a bit confused as to what is going on here? Why is there a change saying that mssql uses tedious v4, when it uses the latest version (v16)?

Plus tedious provides its own typings, so not sure why that would need to be defined here now?

File: types/mssql/package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     ],
     "dependencies": {
         "@types/node": "*",
-        "@types/tedious": "*",
+        "@types/tedious": "^4",
```
Comment (sheetalkamat) [inline types/mssql/package.json:10]:
```suggestion
        "@types/tedious": "*",
```
Please fix errors if any after reverting this change

File: types/tedious-connection-pool/package.json [hunk 2]
```diff
@@ -7,7 +7,7 @@
     ],
     "dependencies": {
         "@types/node": "*",
-        "@types/tedious": "*"
+        "@types/tedious": "^4"
```
Comment (sheetalkamat) [inline types/tedious-connection-pool/package.json:10]:
```suggestion
        "@types/tedious": "*",
```
Please fix errors if any after reverting this change

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/tedious/v4/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{
```
Comment (sheetalkamat) [inline types/tedious/v4/package.json:1]: Also i dont think you need v4 copy at all.

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/tedious/v4/package.json (hunk 3 above)
Comment (lohart13) [reply types/tedious/v4/package.json]: Ok, I've reverted the change now

File: types/tedious-connection-pool/package.json (hunk 2 above)
Comment (lohart13) [reply types/tedious-connection-pool/package.json]: Done

File: types/mssql/package.json (hunk 1 above)
Comment (lohart13) [reply types/mssql/package.json]: Done

File: types/mssql/index.d.ts [hunk 4]
```diff
@@ -206,7 +206,7 @@ export interface config {
     options?: IOptions | undefined;
     pool?: PoolOpts<Connection> | undefined;
     arrayRowMode?: boolean | undefined;
-    authentication?: tds.ConnectionAuthentication | undefined;
+    authentication?: tds.AuthenticationOptions | undefined;
```
Comment (RyanCavanaugh) [inline types/mssql/index.d.ts:209]: Seems like this should be in its own PR?

**Review by RyanCavanaugh — CHANGES_REQUESTED** [review]
